runtime/src/bank.rs

```diff
@@ -105,18 +106,26 @@ pub struct Bank {
 
     /// Slot leader
     leader: Pubkey,
+
+    pub id: u64,
```

Comment:
The `pub` is no good. Also, can you use `Bank::slot_height()` instead?

Reply:
slot_height doesn't work because there can exist multiple banks in the system with the same slot_height and they will then have a shared account state which is a problem.

Reply:
In what scenario is there two banks with the same slot height?

Reply:
How the bank works today or at least at end of the last week is that new_from_parent is called to create the TPU bank. So the TVU and TPU then have banks at the same slot height and are applying transactions to each one.

Reply:
We need to fix that up. A single Bank instance should not change slot heights. I'll play with that. In the meantime, you should be able to move forward assuming that invariant.

Reply:
It's looking pretty busted to me. Hard to tell because that code is buried in the ReplayStage monolith.  We need to get @aeyakovenko's [PR](https://github.com/solana-labs/solana/pull/2852) merged and ensure that the first tick of the TPU bank doesn't sneak into the TVU's bank that we fork from.